### PR TITLE
Updated card styles and added card component 🗃️

### PR DESCRIPTION
**Pull request by mperrotti**
#### Related issues
Fixes https://meetup.atlassian.net/browse/SDS-415

#### Description
Updated card styles based on styles defined by Chris while designing Event and Group home. Moving `card--group` to `mup-web` as part of [WC-73](https://meetup.atlassian.net/browse/WC-73)

#### Screenshots (if applicable)
screen shot 2017-08-22 at 12 31 58 pm

### CODE REVIEW

File: assets/scss/components/_avatar.scss [hunk 1]
```diff
@@ -44,7 +44,7 @@ Class                     | Description
 `avatar--noPhoto`         | Resets `text-indent` for avatars without photos, allowing display of fallback content
 */
 $noPhotoIconOpacity: alpha($C_textHint);
-$avatarBg: #E4E9ED; // change to $C_coolGrayLight once we start using `swarm-constants`
+$avatarBg: $C_coolGrayLight;
```
Comment (mperrotti) [inline assets/scss/components/_avatar.scss]: Not really part of this PR, but I figured I'd update SASS vars after upgrading to a version of `swarm-sasstools` that pulls from `swarm-constants`

File: assets/scss/components/_button.scss [hunk 2]
```diff
@@ -268,8 +268,8 @@ Class                   | Description
 .button--bordered {
 	@include buttonColor(
 		$bgColor: $C_contentBG,
-		$hoverColor: #F6F7F8, //TODO: change to $C_collectionBG once `swarm-constants` is available
-		$activeColor: darken(#F6F7F8, 3%)
+		$hoverColor: $C_lightGray,
```
Comment (mperrotti) [inline assets/scss/components/_button.scss]: Nothing in this file is part of this PR, but I figured I'd update SASS vars after upgrading to a version of `swarm-sasstools` that pulls from `swarm-constants`

File: assets/scss/components/_forms.scss [hunk 3]
```diff
@@ -151,7 +151,7 @@ Faux inputs visually mimic actual input elements.
 }
 
 $toggleSize: 20px;
-$C_toggleHighlight: $C_lightPurple; // replace $C_lightPurple with $C_blue once we start to use `swarm-constants`
+$C_toggleHighlight: $C_blue;
```
Comment (mperrotti) [inline assets/scss/components/_forms.scss]: Not really part of this PR, but I figured I'd update SASS vars after upgrading to a version of `swarm-sasstools` that pulls from `swarm-constants`

File: assets/scss/components/_avatar.scss (hunk 1 above)
Comment (nlaz) [inline assets/scss/components/_avatar.scss]: LOL at `coolGray` 😎

File: src/layout/card.test.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	});
+
+	it('applies variant classes for each variant prop', () => {
+		const variants = [
+			'initialHeight',
+			'hasShadow',
+			'hasHoverShadow'
+		];
+
+		variantTest(Card, CARD_CLASS, variants);
```
Comment (nlaz) [inline src/layout/card.test.jsx]: Probably not part of this PR's scope but this helper method should be updated to use `enzyme` instead of `TestUtils`.

File: assets/scss/components/_card.scss [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+.card--initialHeight {
+	min-height: initial;
+}
+
+.card--hasShadow {
+	@extend %shadow;
+
+	&.card--hasHoverShadow{
```
Comment (nlaz) [inline assets/scss/components/_card.scss]: Can we get a story where both of these variants are used? Not sure what the style here looks like.

File: assets/scss/components/_card.scss [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+		&:hover,
+		&:focus {
+			box-shadow: 0 0 4px 0 $C_border, 0 4px 12px 0 $C_border;
+		}
+
+	}
+
+}
+
+.card--hasHoverShadow {
```
Comment (nlaz) [inline assets/scss/components/_card.scss]: Should these hover states have a default `transition` property? The effect on hover is a little sudden.

Comment (mperrotti) [inline assets/scss/components/_card.scss]: Into this

**Comment by mperrotti** [comment]
I think I'm going to move the `shadowOnHover` styles to some kind of util in `swarm-sasstools`
